### ISSUE TITLE
easyinstall requires pi user, preventing running easyinstall on Ubuntu

### ISSUE THREAD

**Issue by uweseimet** [akuker/RASCSI#384]
rascsi compiles fine with Ubuntu 32 and 64 bit, but the easyinstall script requires a pi user. This dependency on the pi user should be removed, so that easyinstall can be run on a Ubuntu system.
In general, rascsi should not limit the user's choice of the operating system, at least not without a good reason.

After simply removing the initial user id check in easyinstall, easyinstall (option 1) runs fine until here:

```
Compiling the Python protobuf library rascsi_interface_pb2.py...
usermod: group 'pi' does not exist
```

The systemd setup at this point appears to be valid, rascsi is started by systemd after a reboot. The webserver is also up but reports

```

RaSCSI-Web is Starting....
This page will automatically refresh.

First boot and upgrades can take a second while resolving dependencies.

If you're seeing this page for over a minute please check the logs at sudo journalctl -f

```

**Comment by akuker** [comment]
Great discussion guys!

I haven't looked at this code lately, but last time I looked, the RaSCSI service itself needs to run as root, since it controls the GPIO pins via shared memory. Additionally, the RaSCSI service will allow TCP control connections from anywhere.

Going back to the original request, I agree with the idea of being able to use any user id. It was a little hacky to assume that the user will always be 'pi'.

The discussion about moving the web server and/or the rascsi server into a compartment or improving the general security approach to RaSCSI is a good one. But, it probably belongs in a different issue.

**Comment by rdmark** [comment]
This PR should address the matter of any user being able to install and run all components of RaSCSI: https://github.com/akuker/RASCSI/pull/426

@uweseimet The one oddity that I ran into is that rascsi defaulted to ~/ as the default image dir and not ~/images when installing as a non-pi user. In my case, I created a user 'dmark', gave it sudoers privileges, checked out rascsi code to a subdir to the home dir (to test non-standard checkout location) /home/dmark/dev/RASCSI then ran through easyinstall.sh.

However, when inquiring the server you get:

```
$ rasctl -s
rascsi server version: 21.99 (development version)
rascsi log levels, sorted by severity:
  trace
  debug
  info
  warn
  err
  critical
  off
Current rascsi log level: info
Default image file folder: /home/dmark
[...]
```

It wasn't immediately obvious to me the heuristic in the rascsi backend how it picks the default image dir.

**Comment by uweseimet** [comment]
@rdmark Regarding the rules for how the default folder is determined when launching rascsi I suggst to have a look at the rascsi sources. The default setting depends on whether rascsi is started as root user or not, which has to do with backwards compatibility

**Comment by rdmark** [comment]
@uweseimet I found the code, in rascsi_image.cpp and confirmed that "/home/pi/images" is indeed the fallback when rascsi was not launched by a regular user with sudo. Since most users will be running rascsi as a systemd daemon, this is going to be a problem for portability I think.

Perhaps I can add some logic to the webapp that sends a command to rascsi which image dir to use based on the location that the webapp itself is running. But this won't help people who are not using the webapp.

**Comment by uweseimet** [comment]
@rdmark When using systemd the user who installed rascsi is unknown, and I guess that systemd always runs services as root (can you confirm)? System services are not user-aware (except for being aware of root), which is their nature. I don't think a user would be surprised by this, because it's just how the OS works.

**Comment by uweseimet** [comment]
@rdmark easyinstall knows who is running the installation, so maybe you can modify easyinstall so that systemd sets the default folder (when starting rascsi) to the user who runs the installation. You generate the command line for starting the service with systemd anyway, don't you?

**Comment by uweseimet** [comment]
@rdmark Just a general note: A new Raspberry Pi OS was released recently.

**Comment by uweseimet** [comment]
@rdmark I would appreciate having an easyinstall option that does not install any service, but stops after everything was compiled.

**Comment by rdmark** [comment]
> 
> 
> @rdmark easyinstall knows who is running the installation, so maybe you can modify easyinstall so that systemd sets the default folder (when starting rascsi) to the user who runs the installation. You generate the command line for starting the service with systemd anyway, don't you?

That's a good idea. I pushed an update to the script, and it works well in my testing. The PR above should be ready for testing & merging now, I think.

**Comment by rdmark** [comment]
> 
> 
> @rdmark I would appreciate having an easyinstall option that does not install any service, but stops after everything was compiled.

I could for instance add an 'Advanced Users' section at the bottom of the script's menu, with the options to compile rascsi, and perhaps another one for just installing the dependencies + configuration of rascsi-web (since rascsi-web runs just fine as a standalone process if you launch it with start.sh)

**Comment by uweseimet** [comment]
> I could for instance add an 'Advanced Users' section at the bottom of the script's menu, with the options to compile rascsi, and perhaps another one for just installing the dependencies + configuration of rascsi-web (since rascsi-web runs just fine as a standalone process if you launch it with start.sh)

@rdmark An section for advanced users sounds good.

**Comment by rdmark** [comment]
@uweseimet Created https://github.com/akuker/RASCSI/issues/438 for this

Also, the PR above has been merged, so please test that the script works in your Ubuntu environment now!

**Comment by uweseimet** [comment]
@rdmark Compiling and installing the rascsi service worked fine, tested on the current Ubuntu 21.10 64 bit. Only that fact that the binaries are installed in /usr/local/bin instead of a package-specific folder in /opt, like /opt/rascsi, is not as it should be on Linux systems. I suggest to change the installation folder to /opt/rascsi. The target folder would have to be an optional setting, because /opt is the standard location anyway.
I did not check the webservice installation, because it would most likely modify/destroy my existing webserver configuration.
What I just noticed: I did not find the location of the manpages. Where have they been installed? Default location on Linux would be /opt/rascsi/man.
